### DISCUSSION TITLE
Slow download speed with mitmproxy

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6780] [General]
We noticed that the download (not just upload) speed is quite slow with mitmproxy.
In our testing, a 20 MB file took ~20 seconds to download with mitmproxy.
Is there a way to improve the issue?

Here is my mitmproxy version:

Mitmproxy: 10.2.4
Python:    3.10.11
OpenSSL:   OpenSSL 3.2.1 30 Jan 2024
Platform:  macOS-14.4-arm64-arm-64bit

**Comment by Srijan67** [comment]
@changsin You can use `--set stream_large_bodies=${value}` . it is present inside documentation in options section. Value can be anything with appreviation like 2k for 2kb , 2m for 2mb. 
example command is `mitmdump --set stream_large_bodies=2k`
